README.md

```diff
@@ -1,4 +1,1060 @@
+
+If this file does not exist then it will default to an empty project.
+
+### 3.3.1. `inputs`
+
+The [`inputs` managed by IOGX](#312-inputs) merged with the inputs defined in your flake. 
+
+You will also find the `self` attribute here (`inputs.self`).
+
+Note that these inputs have been de-systemized against the current system.
+
+This means that you can use the following syntax:
+```nix
+inputs.n2c.packages.nix2container
+```
+As opposed to:
+```nix 
+inputs.n2c.packages.x86_64-linux.nix2container
+```
+
+In general, you don't want to deal with `system` explicitly, but if you must, you can use [`inputs'`](#332-inputs) instead (note the prime `'` sign).
+
+### 3.3.2. `inputs'`
+
+The [`inputs` managed by IOGX](#312-inputs) merged with the inputs defined in your flake. 
+
+You will also find the `self` attribute (`inputs.self`).
+
+Note that, in contrast to [`inputs`](#331-inputs) above, these inputs have *not* been de-systemized. 
+
+This means that you must always specify the system, as in the following example:    
+```nix
+inputs'.n2c.packages.x86_64-darwin.nix2container
+```
+
+You may want to use this in case you need a Nix value (including a derivation) which is only available on one system, but which can be used safely in the context of another system. 
+
+Instances of this are rare: in general you want to deal with [`inputs`](#331-inputs).
+
+The `inputs/inputs'` notation was stolen from [flake-parts](https://flake.parts).
+
+### 3.3.3. `pkgs`
+
+A `nixpkgs` instantiated against the current system (as found in `pkgs.stdenv.system`), for each of your configured [`systems`](#322-systems), and overlaid with goodies from `haskell.nix` and `iohk-nix`. 
+
+A `nixpkgs` is also available at `inputs.nixpkgs.legacyPackages` or `inputs'.nixpkgs.legacyPackages.${pkgs.stdenv.system}` but those should *not* be used because they don't have the required overlays.
+
+You may reference `pkgs` freely to get to the legacy packages or functions in `pkgs.lib`.
+
+### 3.3.4. `meta`
+
+IOGX will call `haskell.nix.cabalProject'` for each of your configured [`haskellCompilers`](#323-haskellcompilers), with/without cross-compiling according to [`shouldCrossCompile`](#325-shouldcrosscompile), and with and without profiling enabled.
+
+The `meta` field contains that information: `haskellCompiler` tells you the current compiler, `enableProfiling` tells you whether Haskell library and executable profiling is enabled, `enableCross` tells you whether cross compilation is available, while `enableHaddock` is currently always set to `false` (but this will change in future versions). 
+
+With the exception of `enableHaddock`, which is used in some repositories to defer plutus plugin errors, the other `meta` fields are unlikely to be needed, but are exposed anyway.
+
+### 3.3.5. `cabalProjectLocal`
+
+This field will be passed directly to `haskell.nix:cabalProject'`. 
+
+This field is optional and defaults to the empty string.
+
+TODO link to haskell.nix docs.
```

Comment:
Can we fix these TODOs, or get tickets for the remainders?

Reply:
done